src/main/java/com/thebois/models/beings/actions/MoveAction.java

```diff
@@ -0,0 +1,130 @@
+package com.thebois.models.beings.actions;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.Objects;
+
+import com.google.common.eventbus.Subscribe;
+
+import com.thebois.Pawntastic;
+import com.thebois.listeners.events.ObstaclePlacedEvent;
+import com.thebois.models.Position;
+import com.thebois.models.beings.IActionPerformer;
+import com.thebois.models.beings.pathfinding.IPathFinder;
+
+/**
+ * Moves the performer towards a specified goal.
+ */
+class MoveAction implements IAction {
+
+    private final Position destination;
+    private final IPathFinder pathFinder;
+    private boolean canReachDestination = true;
+    private LinkedList<Position> path = new LinkedList<>();
+
+    /**
+     * Instantiates with a destination to move towards.
+     *
+     * @param destination The goal to move to.
+     * @param pathFinder  A way of generating paths to positions.
+     */
+    MoveAction(final Position destination, final IPathFinder pathFinder) {
+        this.destination = destination;
+        this.pathFinder = pathFinder;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(destination);
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) return true;
+        if (!(o instanceof MoveAction)) return false;
+        final MoveAction moveTask = (MoveAction) o;
+        return destination.equals(moveTask.destination);
+    }
+
+    @Override
+    public String toString() {
+        return "MoveAction{" + "destination=" + destination + '}';
+    }
```

Comment:
Do we need this?

Reply:
For debugging tests, yes, 100%. It's very difficult otherwise. Just like with `Position` (which you got to see earlier today).

Reply:
But debugging code in "production" 😬

Reply:
It's **toString**, one of the fundemental methods built into all objects in Java. It's not weird to have an object implementing it, no matter the reason.

Representing an object in text format is something intrinsic to all objects, like how you're able to print an `int` even though it is not made out of text. In the same way, you can't really print `MoveAction` because it is not made out of text. But, like with `int`, you can represent it using text, which is what this override does. 

>debugging code in "production"

This is like not wanting analytics in production because it's not relevant to making the application function. It's an auxiliary function of the application. In the case of analytics, it makes it easier to track behavior and find errors, and in the case of `toString` it makes it easier to find errors, by letting other developers understand why they broke the tests. 

It's like not wanting catch-blocks to print the error because that's debug-code.

So what I'm really trying to argue is that having a string representation of `MoveAction` is a needed functionality, because `MoveAction` is a value object, and when some other developer later on changes some code that breaks `MoveAction`-related code, they'll want error messages saying
`"Expected MoveAction{destination={x=1, y=2}}, but actual is MoveAction{destination={x=10, y=20}}"`
instead of what they would get by default: 
`"Expected MoveAction@2314 but actual is MoveAction@2315"`.
The latter tells the programmer nothing at all, which makes it much harder to debug, while the former makes it incredibly easy to see at a glance what has gone wrong. This is auxiliary functionality that is useful in the same way having tests are.

You wouldn't call having unit tests "debugging in production", would you? I'd argue that this `toString` is just like the unit tests.

Reply:
👍

Reply:
Too annoying to argue, eh? 😅 Sorry about that